### ISSUE TITLE
autoreconf gives warnings on 1.14.7-dev (with autoconf 2.71)

### ISSUE THREAD

**Issue by patricklodder** [dogecoin/dogecoin#3072]
in `src/secp256k1`:

```
configure.ac:28: warning: The macro `AC_PROG_CC_C89' is obsolete.
configure.ac:28: You should run autoupdate.
configure.ac:28: the top level
```

to solve this we'd need to update the subtree, which is anyway a good idea. I'll try to do this with priority because that will need to see lots of exposure - this is consensus critical code.

in the root:

```
configure.ac:454: warning: $as_echo is obsolete; use AS_ECHO(["message"]) instead
lib/m4sugar/m4sh.m4:692: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
build-aux/m4/ax_pthread.m4:88: AX_PTHREAD is expanded from...
configure.ac:454: the top level
```

Should be fixable with an update of [build-aux/m4/ax_pthread.m4](https://github.com/dogecoin/dogecoin/blob/1.14.7-dev/build-aux/m4/ax_pthread.m4)

**Comment by patricklodder** [comment]
Note: these warnings occur on `1.21-dev` too.

**Comment by patricklodder** [comment]
The first part gets addressed for 1.14.7 as part of #3082

**Comment by patricklodder** [comment]
Second part is addressed in #3414

**Comment by patricklodder** [comment]
This has been fixed
